File: IMERA-CS/ImeraWA/ImeraUsuarioBO/AuxiliarBO.cs
using ImeraBaseBO;
using ImeraBaseBO.ImeraServicio;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace ImeraUsuarioBO
{
    public class AuxiliarBO : BaseBO
    {
        public AuxiliarBO():base()
        {
            
        }
        public int insertar(int codigoPucp, string correoPucp,
            string nombre,  string apellidoP,  string apellidoM,
             string idPabellon,  turno _turno,  char activo,string contrasenha,rol _rol)
        {
            return WsCliente.auxiliar_insertar(codigoPucp, correoPucp, nombre, apellidoP, apellidoM, idPabellon, _turno, activo,contrasenha,_rol);
        }

        public int modificar(int idAuxiliar,int codigoPucp, string correoPucp,
            string nombre, string apellidoP, string apellidoM,
             string idPabellon, turno _turno, char activo,string contrasenha)
        {
            return WsCliente.auxiliar_modificar(idAuxiliar,codigoPucp, correoPucp, nombre, apellidoP, apellidoM, idPabellon, _turno, (ushort)activo, contrasenha);
        }
        /*
        public int eliminar(int idEspecialidad)
        {
            return WsCliente.auxiliar_eliminar(idEspecialidad);
        }
        */
        public BindingList<auxiliar> listarTodos()
        {
            auxiliar[] arreglo = WsCliente.auxiliar_listarTodos();
            if (arreglo == null)
                return new BindingList<auxiliar>();
            
            return new BindingList<auxiliar>(arreglo);
        }

        public auxiliar obtenerPorId(int idEspecialidad)
        {
            return WsCliente.auxiliar_obtenerPorId(idEspecialidad);
        }
        /*
        public auxiliar obtenerPorDatosInicioSesion(string DatosUsuario,string contrasenha)
        {
            return WsCliente.auxiliar_obtenerPorIdPorCorreo_Codigo(DatosUsuario,contrasenha);
        }
        */
    }
}

File: IMERA-JAVA/ImeraServidorRMI/ImeraUsuariosBO/src/pe/edu/pucp/imera/usuarios/bo/AuxiliarBO.java
package pe.edu.pucp.imera.usuarios.bo;
import java.util.ArrayList;
import pe.edu.pucp.imera.usuarios.dao.AuxiliarDAO;
import pe.edu.pucp.imera.usuarios.daoImpl.AuxiliarDAOImpl;
import pe.edu.pucp.imera.usuarios.model.Auxiliar;
import pe.edu.pucp.imera.usuarios.model.Rol;
import pe.edu.pucp.imera.usuarios.model.Turno;
import pe.edu.pucp.imera.util.Cifrado;

public class AuxiliarBO{
    
    private final AuxiliarDAO auxiliarDAO;
    
    public AuxiliarBO(){
        this.auxiliarDAO = new AuxiliarDAOImpl();
    }
    
    public Integer insertar(Integer codigoPucp, String correoPucp, String nombre, String apellidoP, 
            String apellidoM, String idPabellon, Turno turno, Character activo,String contrasenha,Rol rol){
        Auxiliar auxiliar = new Auxiliar(codigoPucp, correoPucp, nombre, apellidoP, apellidoM, idPabellon, turno, activo,Cifrado.cifrarMD5(contrasenha),rol);
        return auxiliarDAO.insertar(auxiliar);
    }
    
    public Integer modificar(Integer idAuxiliar, Integer codigoPucp, String correoPucp, String nombre, 
            String apellidoP, String apellidoM, String idPabellon, Turno turno, Character activo,String contrasenha){
        Auxiliar auxiliar = new Auxiliar(idAuxiliar, codigoPucp, correoPucp, nombre, apellidoP, apellidoM, idPabellon, turno, activo,Cifrado.cifrarMD5(contrasenha),null);
        return auxiliarDAO.modificar(auxiliar);
    }
    
    public Integer eliminar(Integer idAuxiliar){
        Auxiliar auxiliar = new Auxiliar();
        auxiliar.setIdAuxiliar(idAuxiliar);
        return auxiliarDAO.eliminar(auxiliar);
    }
    
    public ArrayList<Auxiliar> listarTodos(){
        return auxiliarDAO.listarTodos();
    }
    
    public Auxiliar obtenerPorId(Integer idAuxiliar){
        return auxiliarDAO.obtenerPorId(idAuxiliar);
    }
    //public Auxiliar obtenerPorIdPorCorreo_Codigo(String DatosInicioSesion,String contrasenha){
    //    return auxiliarDAO.obtenerPorIdPorCorreo_Codigo(DatosInicioSesion, Cifrado.cifrarMD5(contrasenha));
    //}
}
